frontend/src/metabase/databases/components/DatabaseEngineList/DatabaseEngineList.tsx

```diff
@@ -0,0 +1,161 @@
+import { useState } from "react";
+import { t } from "ttag";
+
+import { getEngines } from "metabase/databases/selectors";
+import {
+  getEngineLogo,
+  getEngineOptions,
+} from "metabase/databases/utils/engine";
+import { useSelector } from "metabase/lib/redux";
+import {
+  Button,
+  Center,
+  Combobox,
+  Group,
+  Icon,
+  ScrollArea,
+  Stack,
+  Text,
+  useCombobox,
+} from "metabase/ui";
+
+import S from "./DatabaseEngineList.module.css";
+
+export const DatabaseEngineList = ({
```

Comment:
If this component looks similar to https://github.com/metabase/metabase/blob/master/frontend/src/metabase/databases/components/DatabaseEngineField/DatabaseEngineWidget.tsx, that's because it is. The (product+design) idea was to introduce this new component first and then replace the `DatabaseEngineWidget` that's currently used in the DatabaseSetup during the setup flow.